federationtypes_test.go

```diff
@@ -90,38 +94,68 @@ func TestRespStateUnmarshalJSON(t *testing.T) {
 }
 
 func TestRespSendJoinMarshalJSON(t *testing.T) {
+	// we unmarshall and marshall an empty send-join response, and check it round-trips correctly.
 	inputData := `{"state":[],"auth_chain":[],"origin":""}`
 	var input RespSendJoin
 	if err := json.Unmarshal([]byte(inputData), &input); err != nil {
 		t.Fatal(err)
 	}
 
+	want := RespSendJoin{
+		StateEvents: []RawJSON{},
+		AuthEvents:  []RawJSON{},
+		Origin:      "",
+	}
+	if !cmp.Equal(input, want, cmp.AllowUnexported(RespSendJoin{})) {
+		t.Errorf("json.Unmarshal(%s): wanted %+v, got %+v", inputData, want, input)
+	}
+
 	gotBytes, err := json.Marshal(input)
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	got := string(gotBytes)
-
 	if emptyRespSendJoinResponse != got {
-		t.Errorf("json.Marshal(RespSendJoin(%q)): wanted %q, got %q", inputData, emptyRespStateResponse, got)
+		t.Errorf("json.Marshal(%+v): wanted '%s', got '%s'", input, emptyRespSendJoinResponse, got)
 	}
 }
 
-func TestRespSendJoinUnmarshalJSON(t *testing.T) {
-	inputData := `{"state":[],"auth_chain":[],"origin":""}`
+func TestRespSendJoinMarshalJSONPartialState(t *testing.T) {
+	inputData := `{
+        "state":[],"auth_chain":[],"origin":"o1",
+	    "org.matrix.msc3706.partial_state":true,
+        "org.matrix.msc3706.servers_in_room":["s1", "s2"]
+    }`
```

Comment:
nit: strange indentation here. the middle line has a mix of spaces and tabs.

Reply:
thanks. fixed.